src/user/Admin/Tickets/TicketDashboard.js

```diff
@@ -0,0 +1,294 @@
+    this.setState({
+      editorMode: open,
+    });
+  };
+
+  getTicketNotifications = async () => {
+    try {
+      const notifications = (
+        await axios.get(`${BASE_URL}/notification/ticket/user/all`, {
+          cancelToken: this.axiosCancel.token,
+        })
+      ).data.notifications;
+      this.setState({ notifications });
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  handleCreateNewTicket = (newTicket) => {
+    this.setState(
+      {
+        spinner: "Creating new Ticket...",
+      },
+      createNewTicket.bind(this, newTicket)
+    );
+  };
+
+  handleTicketSingleUpdate = (id, update) => {
+    const tickets = [...this.state.all];
+    tickets.forEach((ele) => {
+      if (ele._id === id) {
+        ele[Object.keys(update)[0]] = Object.values(update)[0];
+      }
+    });
+    this.setState({
+      all: [...tickets],
+      filtered: [...tickets],
+    });
+  };
+
+  handleAddTag = (id, tagName) => {
+    const tickets = [...this.state.all];
+    tickets.forEach((ele) => {
+      if (ele._id === id) {
+        ele.tags.push(tagName);
+      }
+    });
+    this.setState({
+      all: [...tickets],
+      filtered: [...tickets],
+    });
+  };
+
+  handleRemoveTag = async (id, tagName) => {
+    const tickets = [...this.state.all];
+    tickets.forEach((ele) => {
+      if (ele._id === id) {
+        ele.tags.splice(ele.tags.indexOf(tagName), 1);
+      }
+    });
+    this.setState({
+      all: [...tickets],
+      filtered: [...tickets],
+    });
+  };
+
+  handleViewTicket = (id) => {
+    this.setState({
+      filtered: [...this.state.all],
+      viewingTicket: id,
+    });
+  };
+
+  deleteTicket = (id) => {
+    this.setState(
+      { spinner: "Deleting Ticket,,,", viewingTicket: null },
+      deleteTicket.bind(this, id)
+    );
+  };
+
+  render() {
+    const { spinner, editorMode, viewingTicket, ticket, all, filtered, notificationDrawer, notifications } = this.state;
+    return (
+      <div className="ticket">
+        <div className="navigation">
+          <Navigation ticket={ticket} />
+        </div>
+        <div className="ticket-details" id="ticket-shadow">
+          <LoadingOverlay
+            text={spinner}
+            active={!!spinner}
+            spinner={<ClockLoader color={"#1A73E8"} />}
+            styles={{
+              spinner: (base) => ({
+                ...base,
+                width: "100px",
+                "& svg circle": {
+                  stroke: "rgba(26, 115, 232, 0.5)",
+                },
+              }),
+            }}
+          >
+            <div className="ticket-description">
+              <div className="dashboard-title">
+                Tickets
+                <Button variant="light" onClick={this.toggleDrawer}>
+                  <NotificationsNoneOutlinedIcon />
+                </Button>
+              </div>
+              {!editorMode && !viewingTicket && (
+                <React.Fragment>
```

Comment:
Better to store in a variable and then just use here, It would be beneficial in debugging the code